### PR TITLE
Sort bins after mass updates

### LINKED ISSUE #739: Check and Re-order bins
We need to check and make sure the bins are increasing in size, and if not re-order them. Un-ordered bins do not work for the coagulation kernel interpolation.

### PR DESCRIPTION
**Pull request by Gorkowski**
## Summary
- add helper `_enforce_increasing_bins` to guarantee distribution is sorted
- run sorting after adding mass
- test that an unsorted particle is reordered

Fixes #739

## Testing
- `pytest -q`

------
https://chatgpt.com/codex/tasks/task_e_68430ebc8dc0832287f4cc796744f671

## Summary by Sourcery

Ensure particle distribution bins remain sorted by radius after mass or concentration updates by introducing an internal sorting helper and invoking it in relevant methods.

Bug Fixes:
- Automatically reorder distribution, concentration, and charge arrays in `add_mass` and `add_concentration` to maintain sorted bins.

Enhancements:
- Introduce `_enforce_increasing_bins` helper to sort bins by increasing radius.

Tests:
- Add test to verify bins, concentrations, and charges are reordered correctly after mass addition.

## Summary by Sourcery

Ensure particle distribution bins remain sorted by increasing radius after mass or concentration updates by introducing a dedicated sorting helper and invoking it automatically in relevant methods.

New Features:
- Add `get_sorted_bins_by_radius` utility to reorder distribution, concentration, and charge arrays by increasing radius
- Introduce `_enforce_increasing_bins` method in the `Particle` class to apply sorting after updates

Bug Fixes:
- Automatically reorder bins, concentrations, and charges in `add_mass` and `add_concentration` to maintain sorted radii

Enhancements:
- Extract bin-sorting logic into a standalone module for reuse

Tests:
- Add tests for `get_sorted_bins_by_radius` covering sorted/unsorted vectors, matrices, and zero-charge scenarios
- Add test to verify bin ordering, concentration, and charge reordering after mass addition

### CODE REVIEW

File: particula/particles/properties/sort_bins.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        - charge : The (optional) charge per particle; can be a scalar or NumPy array.
+
+    Returns:
+        - distribution : The sorted distribution of particle sizes or masses.
+        - concentration : The sorted concentration of each particle size/mass.
+        - charge : The sorted charge of each particle size/mass.
+    """
+    sort_index = np.argsort(radius)
+    sorting_is_needed = not np.array_equal(sort_index, np.arange(radius.size))
+
```
Comment (Gorkowski) [inline particula/particles/properties/sort_bins.py:34]: Do you have opinions on explicit if statement variable names? vs putting the logic right in the if statement. 

I saw Uncle Bob do this, I think it is clearer to read, but then you get these additional statements before each if. I think for complex if statements it is good, but for simple ones don't do it?

File: particula/particles/properties/sort_bins.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    if sorting_is_needed:
```
Comment (Gorkowski) [inline particula/particles/properties/sort_bins.py:35]: Do you have opinions on explicit if statement variable names? vs putting the logic right in the if statement. 

I saw Uncle Bob do this, I think it is clearer to read, but then you get these additional statements before each if. I think for complex if statements it is good, but for simple ones don't do it?
